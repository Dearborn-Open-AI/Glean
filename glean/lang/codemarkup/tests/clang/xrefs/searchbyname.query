query: search.code.SearchByNameAndLanguage _
transform: [gensort, []]
