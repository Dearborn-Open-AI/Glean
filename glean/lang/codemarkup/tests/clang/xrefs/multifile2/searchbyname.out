[
  "@generated",
  {
    "key": {
      "name": "R",
      "entity": {
        "cxx": {
          "defn": {
            "record_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": "R" },
                        "scope": { "global_": { } }
                      }
                    },
                    "kind": { "struct_": { } },
                    "source": {
                      "file": { "key": "b.h" },
                      "lineBegin": 9,
                      "columnBegin": 1,
                      "lineEnd": 9,
                      "columnEnd": 11
                    }
                  }
                },
                "bases": [ ],
                "members": { "key": [ ] }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "name": "T",
      "entity": {
        "cxx": {
          "defn": {
            "record_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": "T" },
                        "scope": { "global_": { } }
                      }
                    },
                    "kind": { "struct_": { } },
                    "source": {
                      "file": { "key": "a.h" },
                      "lineBegin": 9,
                      "columnBegin": 1,
                      "lineEnd": 9,
                      "columnEnd": 11
                    }
                  }
                },
                "bases": [ ],
                "members": { "key": [ ] }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "name": "f",
      "entity": {
        "cxx": {
          "defn": {
            "function_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": { "name": { "key": "f" } } },
                        "scope": { "global_": { } }
                      }
                    },
                    "signature": {
                      "key": { "returns": { "key": "void" }, "parameters": [ ] }
                    },
                    "source": {
                      "file": { "key": "a.h" },
                      "lineBegin": 10,
                      "columnBegin": 1,
                      "lineEnd": 10,
                      "columnEnd": 11
                    }
                  }
                },
                "isInline": false
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "name": "g",
      "entity": {
        "cxx": {
          "defn": {
            "function_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": { "name": { "key": "g" } } },
                        "scope": { "global_": { } }
                      }
                    },
                    "signature": {
                      "key": { "returns": { "key": "void" }, "parameters": [ ] }
                    },
                    "source": {
                      "file": { "key": "a.h" },
                      "lineBegin": 12,
                      "columnBegin": 1,
                      "lineEnd": 15,
                      "columnEnd": 1
                    }
                  }
                },
                "isInline": false
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "name": "h",
      "entity": {
        "cxx": {
          "defn": {
            "function_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": { "name": { "key": "h" } } },
                        "scope": { "global_": { } }
                      }
                    },
                    "signature": {
                      "key": { "returns": { "key": "void" }, "parameters": [ ] }
                    },
                    "source": {
                      "file": { "key": "b.h" },
                      "lineBegin": 11,
                      "columnBegin": 1,
                      "lineEnd": 14,
                      "columnEnd": 1
                    }
                  }
                },
                "isInline": false
              }
            }
          }
        }
      }
    }
  }
]